src/__tests__/embed.spec.ts

```diff
@@ -34,5 +35,111 @@ describe("buildEmbedPlugin", () => {
         propDoesNotExist: "<p>Example initial state</p>",
       });
     });
+
+    it("should not allow fields to be instantiated with an incorrect type", () => {
+      const testEmbed = createNoopEmbed("testEmbed", {
+        prop1: { type: "checkbox", defaultValue: { value: false } },
+      });
+      const { insertEmbed } = buildEmbedPlugin([testEmbed]);
+      insertEmbed("testEmbed", {
+        prop1: { value: true },
+      });
+      insertEmbed("testEmbed", {
+        // @ts-expect-error -- we should not be able to insert a non-existent field
+        prop1: "This shouldn't typecheck",
```

Comment:
The explanation for this error is misleading. The field does exist, it's just the wrong type.